### ISSUE TITLE
[accessibility] Not possible to determine with VoiceOver if a message is a reply

### ISSUE THREAD

**Issue by tuukkao** [signalapp/Signal-iOS#5907]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

While reading messages with VoiceOver it is not possible to tell if a message is in reply to another message. The message gets read out as if it was a regular message, i.e. not a reply. For this reason it is also impossible to know the message this one was replied to. These bugs make this feature essentially unusable for VoiceOver users.

### Steps to reproduce
- Open a conversation, either with another user or a group (this applies to both)
- Reply to a message marking it as such (i.e. using the reply action - I'm a blind Signal user and don't know exactly how this works without VoiceOver, so please forgive me being a bit vague here)
- Turn on voiceOver and navigate to the message you just sent either by swiping left and right or exploring by touch.

**Actual result:** VoiceOver announces the message just as if it was a regular non-reply message, i.e. [your name] sent: [your message]. No information about this message being a reply gets announced.

**Expected result:** VoiceOver should announce this message to be a reply, something like this: [your name] replied: [your message]. Additionally there should be a mechanism for the user to know which message this one refers to. This could be, for example, a rotor action for jumping to the message in question. I strongly discourage announcing the original message before the reply,  since it significantly increases the time to know the content of the actual reply.

### Screenshots

Not applicable.

### Device info

**Device**: iPhone SE 2020

**iOS version**: 18.1

**Signal version:** 7.34

### Link to debug log

I believe this is not applicable, but I'm happy to provide one if needed.

**Comment by tuukkao** [comment]
I just saw that in the most recent Signal iOs release as of this writing this issue has been partly fixed. Now VoiceOver tells whether a message is a reply or not. Thank you for that!

However, it's still not possible to know what message this one was in reply to. I think that is an essential bit of information to truly make this feature useful. See the issue description above for a suggestion on how this could be implemented.

**Comment by Simon818** [comment]
This has been partially fixed in recent updates, but there's still no way to tell which message is being replied to. Sometimes it's possible to tell from context, other times it's not. I would recommend including this in the VoiceOver hint for the message, which is either read automatically or after tapping once with three fingers; or possibly in the message details screen. It's not something we'll need with every message, but when someone replies to a message and the reply is ambiguous, it's frustrating not to have the same access to information.

**Comment by Simon818** [comment]
This is in the same state as most recently discussed: It's still not possible to tell which message a reply is referencing. Pretty frustrating in busy group chats. I am running beta builds.

**Comment by MarlowBrown** [comment]
Hey @Simon818. I'll take a look at this soon and see what I can do.

**Comment by MarlowBrown** [comment]
Keep open

**Comment by Simon818** [comment]
I created #6134 in an attempt to propose a viable solution to all of the accessibility problems surrounding composite messages, including the outstanding issue of not being able to read the quoted message in a reply. There are other related issues, such as not being able to interact with a link or an attachment properly.

Since the original issue was being able to tell \**whether*\* a message is a reply, and we can now do that, I'm going to suggest moving discussions there. I am not a developer, so if someone who knows the code better disagrees that these should all be lumped into one issue, I'll defer to them.

**Comment by MarlowBrown** [comment]
Keep open

**Comment by MarlowBrown** [comment]
Keep open

**Comment by Simon818** [comment]
This has actually been fixed in a recent update. I'm fairly sure it's out of beta now. You now hear the reply, then the original message.
The only complaint I have is that the reply is not rendered on a separate line. It should be possible to quickly scroll down to it without listening through the whole message.
But it's quite a long way off from where we were when this issue was filed, so I appreciate it.

**Comment by tuukkao** [comment]
Yes, this issue has been fixed in public builds for at least a month. I consider this to be fixed in its original scope.
